src/Collection.php

```diff
@@ -198,10 +197,6 @@ public function empty(): CollectionInterface
      */
     public function slice(int $offset, ?int $length = null): CollectionInterface
     {
-        if ($offset >= count($this->items)) {
-            throw new RuntimeException('Offset exceeds the maximum value.');
```

Comment:
`array_slice([], 2, 1)` works and return `[]`.

I propose to allow this in order to avoid throwing exception in slice method (or you'll need to add `@throws RuntimeException` on every other method using slice).

Reply:
This is fine for the rather generic class `Collection`. However, it has an effect on the function `Image::sliceAnimation()`, which throws rather incomprehensible exceptions by GD/Imagick, which are rather confusing for someone who does not know the library inside out.

For this reason, I have reintegrated the check, but moved it to the `SliceAnimationModifier` classes.